src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,48 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  align-items: center;
+  gap: $gap-size;
+  padding: $padding-size;
+  width: $calendar-width;
+
+  @for $index from 28 through 31 {
+    &--month-length--#{$index} {
+      & :nth-child(n + #{$index + 1}) {
+        display: none;
+      }
+    }
+  }
+
+  @each $week-day, $count in $week-days {
+    &--start-day--#{$week-day} {
+      & :first-child {
+        margin-left: $count * $day-cell-side + $count * $gap-size;
+      }
+    }
+  }
+}
+
+.calendar__day {
+  @extend %display-flex-center;
+  width: $day-cell-side;
+  height: $day-cell-side;
+
+  background-color: #eee;
+  border: 1px solid #000;
+
+  transition: 0.5s;
```

Comment:
Better to define which exact properties will be animated, because if you not define it, it will be applied to every property that can be animated